external-crates/move/crates/move-vm-config/src/runtime.rs

```diff
@@ -56,24 +62,23 @@ impl Default for VMRuntimeLimitsConfig {
     }
 }
 
-#[cfg(debug_assertions)]
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct VMProfilerConfig {
-    /// Base path for files
-    pub base_path: std::path::PathBuf,
+    /// User configured full path override
+    pub full_path: std::path::PathBuf,
     /// Whether or not to track bytecode instructions
     pub track_bytecode_instructions: bool,
     /// Whether or not to use the long name for functions
     pub use_long_function_name: bool,
 }
 
-#[cfg(debug_assertions)]
-impl std::default::Default for VMProfilerConfig {
-    fn default() -> Self {
-        Self {
-            base_path: std::path::PathBuf::from("."),
-            track_bytecode_instructions: false,
-            use_long_function_name: false,
+#[cfg(feature = "gas-profiler")]
+impl VMProfilerConfig {
+    pub fn get_default_config_if_enabled() -> Option<VMProfilerConfig> {
+        if *PROFILER_ENABLED {
+            Some(VMProfilerConfig::default())
```

Comment:
This sets the `full_path` to the default value for `PathBuf` -- that seems like a change in behaviour to how things worked before (in terms of what the default path is that the profile gets put in) -- I might be missing something happening in another file though.

Reply:
I thought that was what you requested? The full path is resolved earlier and passed into the profiler? According to my testing the output gets written the same as before

Reply:
My comment was about moving the logic for how the default value of the config gets calculated higher up the call stack so that you could always assume that `VMProfilerConfig` had a non-optional `full_path`, but I still wanted the name of the profile output to have the same format as it always did, and IIUC this change makes it so that the name of the profile output would be different in the case when it's enabled via the environment variable? But like I said, I may be missing something -- I'll pull the branch and play with it to understand better.